src/iop/hlreconstruct/opposed.c

```diff
@@ -272,7 +272,27 @@ static float *_process_opposed(
       chrominance[c] = img_oppchroma[c];
     if(!img_oppclipped && !keep)
     {
-      dt_iop_copy_image_roi(output, input, 1, roi_in, roi_out, 0);
+      if(roi_in->width <= roi_out->width && roi_in->height <= roi_out->height)
+        dt_iop_copy_image_roi(output, input, 1, roi_in, roi_out, 0);
+      else
+      {
+#ifdef _OPENMP
+#pragma omp parallel for default(none) \
+  dt_omp_firstprivate(output, input, roi_in, roi_out) \
+  schedule(static) collapse(2)
+#endif
+        for(size_t row = 0; row < roi_out->height; row++)
```

Comment:
Wouldn't it make more sense to put this code in the `if` branch inside `dt_iop_copy_image_roi` that is currently saying it is not implemented?  That would also fix the problem if any future code wants to call the function with a larger output RoI.

Reply:
I am not sure what the fill-in values for out of roi_in would be. Always zero?

Reply:
The intent of the `zero_pad` arg was that the fill-in should be zeros if TRUE.  Probably its type should be changed to gboolean if we don't just remove it and always use zero fill (I'm not sure there's anywhere a non-zero fill or leaving the extra pixels as-is would make sense).

Reply:
I think we should always fill with something to avoid NaN data.

So either always fill with zero or change the int to float and check all callers.

Reply:
> I think we should always fill with something to avoid NaN data.

Agreed.  And since I don't think there's anywhere that a nonzero fill makes sense, the thing to do is remove that final arg.  I checked, and of 29 calls, the majority use TRUE, most of the remainder use 0,  and two use 1.  Implementing the `if` branch code here and putting the removal of that arg in a followup PR might be the cleanest approach.

Reply:
Fine with me.